malcolm/modules/ADCore/parts/hdfwriterpart.py

```diff
@@ -41,8 +41,8 @@
 APartRunsOnWindows = APartRunsOnWindows
 
 
-def greater_than_zero(v: int) -> bool:
-    return v > 0
+def greater_than_value(v: int, compare_value: int = 0) -> bool:
+    return v > compare_value
```

Comment:
If we write as
```suggestion
def greater_than_value(compare_value: int = 0) -> Callable[[int], bool]:
    def greater_than(v: int) -> bool:
        return v > compare_value
    greater_than.__name__ = f"greater_than_{compare_value)}"
    return greater_than
```
then can we avoid passing kwargs to `When`?

Reply:
I wasn't a huge fan of my implementation either, and do like this one. However resetting the array counter on seek to zero avoids needing to change the method from the original which is even nicer!